### DISCUSSION TITLE
The definition of UGRD_VGRD statistic

### DISCUSSION THREAD

**Discussion by Ho-ChunHuang-NOAA** [dtcenter/METplus#1591] [Statistical Computation] [answered]
Hi, when we use the UGRD and VGRD for poist_stat.  There is a variable name called UGRD_VGRD.  Can you tell me what is it?  I was told it is vector wind, thus from the EMC VSDB document (see below) it can not be used to compute the bias for wind speed = sqrt(u*u+v*v)?

Is it X6 and X7

  | #    X6=MEAN[uf*uf+vf*vf], X7=MEAN[ua*ua+va*va]
  | #    where  uf and vf are forecast winds,  and ua and va are analysis/observed winds,

Mean biases:    bias=sqrt[ X6 ] - sqrt[ X7 ]

or  V_f and V_a?

 V_f and V_a are **vector winds**, V_f=uf(i)+vf(j), V_a=ua(i)+va(j).

**Comment by jprestop** [comment]
Hi @Ho-ChunHuang-NOAA.  

Thank you for your question.  I see you have some questions about wind vectors.  Please read through [Appendix G Vectors and Vector Statistics](https://met.readthedocs.io/en/latest/Users_Guide/appendixG.html?highlight=wind) of the MET User's Guide and see if it answers your questions.  Please follow up with any questions or let us know that you're all set after taking a look.

**Comment by Ho-ChunHuang-NOAA** [comment]
My guess is X6=FS_RMS\*FS+RMS and X7 is OS_RMS\*OS_RMS.  Although, the definition of FS_RMS and OS_RMS are the root-mean-square values of the forecast and observed wind speeds.  It is like the average of the (forecasted/observed) wind speed at each observation sites, am I correct?

V_f is vector wind of Fi  and V_a is vector wind of Oi.  It is more difficult to understand.  Can you show an example of comparing between OBAR_SPEED and FBAR_SPEED?

**Reply by jprestop** [reply]
Hi @Ho-ChunHuang-NOAA.  Thank you for letting us know that you still have questions.  I need to defer to @JohnHalleyGotway's expertise.  @JohnHalleyGotway, could you please help answer Ho-Chun's questions?

**Reply by JohnHalleyGotway** [reply]
@Ho-ChunHuang-NOAA, I see that your original question was about the string "UGRD_VGRD". You'll find that that string appears in the FCST_VAR and OBS_VAR output columns for the stat line types for vector winds... which are the VL1L2, VAL1L2, and VCNT line types. There's not much to say about that. Its just a naming convention from the original NCEP VSDB software that we adopted for MET as well. It just implies that the vector wind line types are computed using both the U and V wind components.

To be clear, there is no *STATISTIC* named UGRD_VGRD in the MET output. Instead UGRD_VGRD is the *VARIABLE NAME* associated with the vector partial sums and statistics listed in the [VL1L2 line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id21), [VAL1L2 line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id22), and [VCNT line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id23).

So the first step is clearing up that confusion. What are you looking at that implies that UGRD_VGRD is a statistic that corresponds to some equation? Is it METviewer? If possible, please send a screen shot to illustrate.

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi,

The line type in my statistic output is VL1L2.

It is my mistake that UGRD_VGRD indeed is the var name.  My questions is
related tot he statistic name in the drop-down menu that includes VCNT_*,
VL1L2_*, ...etc.  Which one I choose to show the difference bwteen modeled
and observed wind in relative values (time-series). difference (ME,
BIAS,..., RMSE), and taylor's diagram.

Note this is the first time I look at the wind speed verification.
Different from TMP or DPT, it is not straight forward to use FBAR, OBAR,
ME, RMSE, Tayplor's diagram.  Now, I need to understand the ifference
between VCNT_* VL2L2_*, but the statistic may be available toone but not
for another one.

Ho-Chun Huang, Ph.D.

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, John:


It took me some time to go through the equation of each MET calculated or
METViewer derived statistic.

Please confirm that for a selected domain, e.g., NEC, and use definition of
https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id21

VL1L2_BIAS=UVFFBAR - UVOOBAR

calcVL1L2_BIAS = function(d){
bias = sqrt(d$uvffbar) - sqrt(d$uvoobar);
return ( round(bias, digits=5)) ;
}
my interpretation is that VL1L2_BIAS is the difference of mean domain wind
speed between forecasted and observed  (or bias=Fa-Oa; as Fa and Oa in
https://met.readthedocs.io/en/latest/Users_Guide/appendixG.html )  I
actually need to find what is the d$uvffbar and d$uvoobar. Do they mean
domain average (Fa) or an array over obs sites of selected domains (sfi)?


While  VL1L2_MSVE as in
https://met.readthedocs.io/en/latest/Users_Guide/appendixG.html is the mean
difference between forecast wind and observation wind) within domain NEC.


I know you might be confused about the terminology I used since I reference
it over several places .  Please let me know and I will try to clarify.

The other question, for your experience, what is the best quantity
(statistics) to demonstrate how well the model predicts the wind speed? You
probably know best about the wind speed impact on the predicted
meteorology.  In air quality application, wind speed is the criteria
whether we will have dust emission.  It also determines whether the
photochemical reaction can be promoted (clam wind) or depressed (windy).
There are also other air quality impacts depending on the stronger or
weaker wind speed prediction.

Ho-Chun Huang, Ph.D.

**Reply by JohnHalleyGotway** [reply]
@Ho-ChunHuang-NOAA, I also find the vector statistics to be somewhat overwhelming and confusing. It's amazing how much the handling of 2 dimensional vectors complicates the statistics when compared to 1 dimensional scalars (like temperature and pressure).

The good news is that wind speed really is a scalar quantity and can be verified as such in MET. If all you actually care about is wind speed, not direction or other vector stats, you could simplify the approach by verifying forecast wind speed against observed wind speed using continuous statistics in the CNT line type. Then you'd use RMSE, MSE, and so on, as you already do for other scalar variables.

However, this would rely on wind speed actually being available in your point observations. If using PREPBUFR, you could configure pb2nc to derive WIND speed from the U and V components. If you request WIND speed from the a GRIB1 or GRIB2 file, MET will automatically find the UGRD and VGRD components and derive wind speed from them on the fly.

So how would you like to proceed? Verify wind speed as a scalar field? Or continue diving in more deeply into the vector stats?

**Comment by Ho-ChunHuang-NOAA** [comment, marked as answer]
Hi, John:

As a response in a separate email, I have no further thought on the wind
comparison.   Initially, I only want to compare wind speed sqrt(u^2+v^2) at
each observation site between modeled and observed.  Compute the difference
(ME or RMSE) and average over the selected domain (VX_domain).

But I notice you have more wind statistics defined beyond my
understanding.  Through this exercise, I found that VCNT_RMSEV is what I
want as described above.

While VL1L2_BIAS is the difference of mean regional wind speed between
modeled and observed.

Actually, I reach my limit to dig deeper in the wind, so I am going to stop
here.


Ho-Chun Huang, Ph.D.
